libraries/classes/Server/UserGroups.php

```diff
@@ -74,96 +72,75 @@ public static function getHtmlForListingUsersofAGroup($userGroup)
-                    . '">'
-                    . Generator::getIcon('b_usrlist', __('View users'))
-                    . '</a>';
-                $html_output .= '&nbsp;&nbsp;';
-                $html_output .= '<a class="" href="' . Url::getFromRoute('/server/user-groups') . '" data-post="'
-                    . Url::getCommon(
-                        [
-                            'editUserGroup' => 1,
-                            'userGroup' => $groupName,
-                        ],
-                        ''
-                    )
-                    . '">'
-                    . Generator::getIcon('b_edit', __('Edit')) . '</a>';
-                $html_output .= '&nbsp;&nbsp;';
-                $html_output .= '<a class="deleteUserGroup ajax"'
-                    . ' href="' . Url::getFromRoute('/server/user-groups') . '" data-post="'
-                    . Url::getCommon(
-                        [
-                            'deleteUserGroup' => 1,
-                            'userGroup' => $groupName,
-                        ],
-                        ''
-                    )
-                    . '">'
-                    . Generator::getIcon('b_drop', __('Delete')) . '</a>';
-                $html_output .= '</td>';
+                $userGroupVal['editUsersUrl'] = Url::getCommon(
+                    [
+                        'editUserGroup' => 1,
+                        'userGroup' => $groupName,
+                    ],
+                    ''
+                );
+                $userGroupVal['editUsersIcon'] = Generator::getIcon('b_edit', __('Edit'));
 
-                $html_output .= '</tr>';
+                $userGroupVal['deleteUsersUrl'] = Url::getCommon(
+                    [
+                        'deleteUserGroup' => 1,
+                        'userGroup' => $groupName,
+                    ],
+                    ''
+                );
+                $userGroupVal['deleteUsersIcon'] = Generator::getIcon('b_drop', __('Delete'));
+                $userGroupsValues[] = $userGroupVal;
             }
-
-            $html_output .= '</tbody>';
-            $html_output .= '</table>';
-            $html_output .= '</form>';
         }
+        $addUserUrl = Url::getFromRoute('/server/user-groups', ['addUserGroup' => 1]);
+        $addUserIcon = Generator::getIcon('b_usradd');
         $GLOBALS['dbi']->freeResult($result);
-
-        $html_output .= '<div class="row"><fieldset id="fieldset_add_user_group">';
-        $html_output .= '<a href="' . Url::getFromRoute('/server/user-groups', ['addUserGroup' => 1]) . '">'
-            . Generator::getIcon('b_usradd')
-            . __('Add user group') . '</a>';
-        $html_output .= '</fieldset></div>';
-
-        return $html_output;
+        $template = new Template();
+        return  $template->render('server/user_groups/user_groups', [
+            'action' => $action,
+            'hidden_inputs' => isset($hidden_inputs)?$hidden_inputs:'',
```

Comment:
```suggestion
            'hidden_inputs' => isset($hidden_inputs) ? $hidden_inputs : '',
```

Do not forget to run `composer run phpcbf` before committing

Reply:
phpcbf gave me tons of fixed files however i pushed only the ones i am working on

Reply:
quite strange
did you `composer update` before ?

Reply:
> quite strange
> did you `composer update` before ?

no